src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue

```diff
@@ -1,114 +1,183 @@
+    type?: ContextMenuType;
+    disabled?: boolean;
+    link?: string;
+    target?: string;
+    icon?: string;
+}
 
 interface Props {
     variableName: string;
-    selected?: Array<MenuItem>;
-    variableSchema?: DashboardVariableSchemaProperty;
+    defaultSelected?: string| string[];
+    variableOptions: string[];
+    selectionType: VariableSelectionType;
+}
+interface EmitFn {
+    (e: 'change', value: string|string[]): void;
 }
 
 const props = defineProps<Props>();
+const emit = defineEmits<EmitFn>();
+
 
 const state = reactive({
-    visibleMenu: false,
-    varialbeType: computed(() => props.variableSchema?.variable_type ?? 'MANAGED'),
-    selectionType: computed(() => props.variableSchema?.selection_type ?? 'SINGLE'),
+    targetRef: null,
+    contextMenuRef: null,
     searchText: '',
-    // TODO: Must be filtered by searchText & reordered by Fn in context menu controller
-    menu: [
-        { name: 'spaceone', label: 'SpaceONE Dev' },
-        { name: 'test1', label: 'Test-1' },
-        { name: 'test2', label: 'Test-2' },
-        { name: 'test3', label: 'Test-3' },
-        { name: 'test4', label: 'Test-4' },
-    ] as MenuItem[],
-    selected: props.selected ?? [
-        { name: 'spaceone', label: 'SpaceONE Dev' },
-        { name: 'spaceone', label: 'SpaceONE Dev' },
-        { name: 'spaceone', label: 'SpaceONE Dev' },
-    ] as MenuItem[],
+    selected: [] as MenuItem[],
+    options: [] as MenuItem[],
 });
+
+const {
+    targetRef,
+    contextMenuRef,
+    selected,
+    options,
+} = toRefs(state);
+
 const {
     visibleMenu,
-    // showContextMenu,
+    showContextMenu,
     hideContextMenu,
     // focusOnContextMenu,
-    // reorderMenuBySelection,
-    // fixedMenuStyle,
+    reorderMenuBySelection,
+    fixedMenuStyle,
+    menu,
 } = useContextMenuController({
-    targetRef: ref(null),
-    contextMenuRef: ref(null),
+    targetRef,
+    contextMenuRef,
     useReorderBySelection: true,
     useFixedStyle: true,
-    menu: ref([]),
+    menu: options,
 });
-// const hideContextMenu = () => {
-//     state.visibleMenu = false;
-// };
-const showContextMenu = (e: MouseEvent) => {
-    state.visibleMenu = !state.visibleMenu;
-    e.stopPropagation();
+
+// event
+const handleClearSelected = () => {
+    state.selected = [];
 };
+const handleChangeVisible = () => {
+    if (visibleMenu.value) {
+        hideContextMenu();
+    } else {
+        reorderMenuBySelection(selected.value);
+        showContextMenu();
+    }
+};
+// TODO: search text binding
 const handleChangeContextMenuInput = (value: string): void => {
-    state.searchText = value;
+    // state.searchText = value;
 };
 
+// reconvert to string | string[]
+watch(() => state.selected, () => {
+    let reconvertedSelected;
+    if (props.selectionType === 'SINGLE') {
+        reconvertedSelected = state.selected[0].name;
+    } else reconvertedSelected = state.selected.map((d) => d.name);
+    emit('change', reconvertedSelected);
+});
```

Comment:
nitpick: I prefer to use first argument of watch callback.
That's because it's much clearer and brief :)

If the target to watch is computedRef, getter runs every time it's been read.
It's pretty inefficient.
Also, of course that won't happen, but programmatically, there is no guarantee that the return value will always be the same. 
(I know that Vue guarantees... but programmatically, it does not. That means, it's possible to misunderstand that this part is causing a bug.)

```suggestion
watch(() => state.selected, (selected) => {
    let reconvertedSelected;
    if (props.selectionType === 'SINGLE') {
        reconvertedSelected = selected[0].name;
    } else reconvertedSelected = selected.map((d) => d.name);
    emit('change', reconvertedSelected);
});
```